opennlp-tools/src/main/java/opennlp/tools/ml/model/OnePassDataIndexer.java

```diff
@@ -39,141 +39,175 @@
-   *          an <code>EventStream</code> value
-   * @param predicatesInOut
-   *          a <code>TObjectIntHashMap</code> value
-   * @param cutoff
-   *          an <code>int</code> value
-   * @return a <code>TLinkedList</code> value
-   */
-  private LinkedList<Event> computeEventCounts(ObjectStream<Event> eventStream,
-      Map<String, Integer> predicatesInOut, int cutoff) throws IOException {
-    Set<String> predicateSet = new HashSet<>();
-    Map<String, Integer> counter = new HashMap<>();
-    LinkedList<Event> events = new LinkedList<>();
-    Event ev;
-    while ((ev = eventStream.read()) != null) {
-      events.addLast(ev);
-      update(ev.getContext(), predicateSet, counter, cutoff);
-    }
-    predCounts = new int[predicateSet.size()];
-    int index = 0;
-    for (Iterator<String> pi = predicateSet.iterator(); pi.hasNext(); index++) {
-      String predicate = pi.next();
-      predCounts[index] = counter.get(predicate);
-      predicatesInOut.put(predicate, index);
-    }
-    return events;
-  }
-
-  protected List<ComparableEvent> index(LinkedList<Event> events,
-      Map<String, Integer> predicateIndex) {
-    Map<String, Integer> omap = new HashMap<>();
-
-    int numEvents = events.size();
-    int outcomeCount = 0;
-    List<ComparableEvent> eventsToCompare = new ArrayList<>(numEvents);
-    List<Integer> indexedContext = new ArrayList<>();
-
-    for (int eventIndex = 0; eventIndex < numEvents; eventIndex++) {
-      Event ev = events.removeFirst();
-      String[] econtext = ev.getContext();
-      ComparableEvent ce;
-
-      int ocID;
-      String oc = ev.getOutcome();
-
-      if (omap.containsKey(oc)) {
-        ocID = omap.get(oc);
-      } else {
-        ocID = outcomeCount++;
-        omap.put(oc, ocID);
-      }
-
-      for (String pred : econtext) {
-        if (predicateIndex.containsKey(pred)) {
-          indexedContext.add(predicateIndex.get(pred));
-        }
-      }
-
-      // drop events with no active features
-      if (indexedContext.size() > 0) {
-        int[] cons = new int[indexedContext.size()];
-        for (int ci = 0; ci < cons.length; ci++) {
-          cons[ci] = indexedContext.get(ci);
-        }
-        ce = new ComparableEvent(ocID, cons);
-        eventsToCompare.add(ce);
-      } else {
-        System.err.println("Dropped event " + ev.getOutcome() + ":"
-            + Arrays.asList(ev.getContext()));
-      }
-      // recycle the TIntArrayList
-      indexedContext.clear();
-    }
-    outcomeLabels = toIndexedStringArray(omap);
-    predLabels = toIndexedStringArray(predicateIndex);
-    return eventsToCompare;
-  }
+	public OnePassDataIndexer(){}
+
+	@Override
+	public void index(ObjectStream<Event> eventStream) throws IOException{
+		int cutoff=parameters.getIntParam(CUTOFF_PARAM, CUTOFF_DEFAULT);
+		boolean sort=parameters.getBooleanParam(SORT_PARAM, SORT_DEFAULT);
+		
+		Map<String, Integer> predicateIndex = new HashMap<>();
+		LinkedList<Event> events;
```

Comment:
This is code that was originally in the constructor that I moved into the index method.  I will be happy to change it to a List<Event>.